week-4/Homework/mandatory/2-writers.js

```diff
@@ -1,44 +1,97 @@
 
-"Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
+Did you know you can also have an Array of Objects? You might think "This is madness!" but in everyday coding life
+it is quite frequent combination. Just think about it what benefits we can get from this construct.
 
-Here is the array:
+Object lets you store multiple values in a single variable, then you can store complex objects in an array.
+Let's assume you have list of data about people names and their birthday and you would like to print each name
+with corresponding birthdays together. Storing these information in different arrays and then pairing them up
+makes the iteration unnecessarily complicated, code will be less intuitive, needs extra cognitive effort to
+reason about and last but not least it can be error-prone for example you pick up the wrong birthday to a name.
 
+In this exercise you will practice how to access to Objects stored in Array and to their properties.
+You already know different ways how to loop through Arrays, it won't be different in this case too.
+The only extra is that you have to use values inside Objects.
 */
 
+// We've created an array of objects for you here:
 let writers = [
   {
     firstName: "Virginia",
     lastName: "Woolf",
     occupation: "writer",
     age: 59,
-    alive: false
+    alive: false,
   },
   {
     firstName: "Zadie",
     lastName: "Smith",
     occupation: "writer",
     age: 41,
-    alive: true
+    alive: true,
   },
   {
     firstName: "Jane",
     lastName: "Austen",
     occupation: "writer",
     age: 41,
-    alive: false
+    alive: false,
   },
   {
     firstName: "bell",
     lastName: "hooks",
     occupation: "writer",
     age: 64,
-    alive: true
-  }
+    alive: true,
+  },
 ];
 
 /*
-If you want an extra challenge, only `console.log()` the writers that are alive.
+Exercise 1:
+  Loop through the Array, and for each object, `console.log()` out the sentence and
+  insert corresponding values to the place holder that are indicated in courle braces:
+  "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
+*/
+function introducing() {
+  for (let i = 0; i < writers.length; i++) {
+    console.log(
+      `Hi, my name is ${writers[i].firstName} ${writers[i].lastName}. I am ${writers[i].age} years old, and work as a ${writers[i].occupation}.`
+    );
+  }
+}
+introducing();
+/*
+Exercise 2:
+  Only `console.log()` the writers who are in their 40s with us anymore with the sentence:(meaning between 40 and 49)
+  "Writer {firstName} {lastName} died at {age} years old."
 */
+function getAgeDied() {
+  let diedAge = writers.filter(
+    (p) => p.age >= 40 && p.age <= 49 && p.alive === true
+  );
+  // console.log(diedAge);
+  for (let n = 0; n < diedAge.length; n++) {
+    console.log(
+      `Writer ${diedAge[n].firstName} ${diedAge[n].lastName} died at ${diedAge[n].age} years old.`
+    );
+  }
+  return diedAge;
+}

```

Comment:
On this part of the exercise, you have an array of Object and you just need to apply the appropriate array methods straight away.